### PR TITLE
Extend tile element size to 16 bytes

### PR DESCRIPTION
**Pull request by Gymnasiast**
A rough, first version. Importing RCT1/RCT2 saves works, but autosaving creates a corrupted version. I'll need some help with that.

### CODE REVIEW

File: src/openrct2/world/TileElement.h [hunk 1]
```diff
@@ -65,6 +65,8 @@ struct TileElementBase
     uint8_t base_height;      // 2
     uint8_t clearance_height; // 3
 
+    uint8_t testPad[8];
```
Comment (IntelOrca) [inline src/openrct2/world/TileElement.h]: This struct shouldn't have any padding. Padding should be added to all the other structs.

Comment (Gymnasiast) [reply src/openrct2/world/TileElement.h]: It was just a test, but I will change it.

File: src/openrct2/rct2/S6Exporter.cpp [hunk 2]
```diff
@@ -1326,6 +1323,173 @@ void S6Exporter::ExportMapAnimations()
...
 
+void S6Exporter::ExportTileElements()
+{
+    for (uint32_t index = 0; index < RCT2_MAX_TILE_ELEMENTS; index++)
+    {
+        auto src = &gTileElements[index];
+        auto dst = &_s6.tile_elements[index];
+        if (src->base_height == 0xFF)
+        {
+            std::memcpy(dst, src, sizeof(*src));
```
Comment (duncanspumpkin) [inline src/openrct2/rct2/S6Exporter.cpp]: What is this doing? Won't this be too big?

Comment (Gymnasiast) [reply src/openrct2/rct2/S6Exporter.cpp]: It copies "null" tile elements over (mirroring S6Importer). It was indeed too big - I have corrected that now

File: src/openrct2/world/TileElement.h [hunk 3]
```diff
@@ -83,6 +83,7 @@ struct TileElementBase
 struct TileElement : public TileElementBase
 {
     uint8_t pad_04[4];
+    uint8_t pad_08[8];
```
Comment (duncanspumpkin) [inline src/openrct2/world/TileElement.h]: I think the padding should **only** be added to `TileElement` and not to any of the others. All of the other structs should 16 or less bytes. `TileElement` is the only one that needs to be `16`. That way we don't have to clutter the code base with pragmas. Its how we have done the sprites (kind of they use a big union).

Comment (Gymnasiast) [reply src/openrct2/world/TileElement.h]: That will cause the assertions of the other tile element types to fail. Not an option.

Comment (IntelOrca) [reply src/openrct2/world/TileElement.h]: @duncanspumpkin While the others don't need to be 16 bytes, it is safer. For example is I were do `SurfaceElement elements[5];` I would probably want that to be 5 * 16 bytes.

**Comment by duncanspumpkin** [comment]
Did you check the tile inspector in multiplayer and increment the network version?

**Comment by ZehMatt** [comment]
Luckily it just refuses to load the map and disconnects currently, but a network bump would have been good.

**Pull request merged**